File: src/main/java/org/oewntk/model/Serialize.java
/*
 * Copyright (c) $originalComment.match("Copyright \(c\) (\d+)", 1, "-")2021. Bernard Bou.
 */

package org.oewntk.model;

import java.io.*;

public class Serialize
{
	static public void serializeModel(final Model model, final File file) throws IOException
	{
		try (OutputStream os = new FileOutputStream(file))
		{
			serializeModel(os, model);
		}
	}

	public static void serializeModel(final OutputStream os, final Model model) throws IOException
	{
		serialize(os, model);
	}

	static public void serializeCoreModel(final CoreModel model, final File file) throws IOException
	{
		try (OutputStream os = new FileOutputStream(file))
		{
			serializeCoreModel(os, model);
		}
	}

	public static void serializeCoreModel(final OutputStream os, final CoreModel model) throws IOException
	{
		serialize(os, model);
	}

	private static void serialize(final OutputStream os, final Object object) throws IOException
	{
		try (ObjectOutputStream oos = new ObjectOutputStream(os))
		{
			oos.writeObject(object);
		}
	}
}

File: src/main/java/org/oewntk/model/Serialize.kt
package org.oewntk.model

import java.io.*

/**
 * Serialize models
 */
object Serialize {
	/**
	 * Serialize model to file
	 *
	 * @param model model
	 * @param file  file
	 * @throws IOException io exception
	 */
	@Throws(IOException::class)
	fun serializeModel(model: Model, file: File) {
		FileOutputStream(file).use { os ->
			serializeModel(os, model)
		}
	}

	/**
	 * Serialize model to output stream
	 *
	 * @param model model
	 * @param os    output stream
	 * @throws IOException io exception
	 */
	@Throws(IOException::class)
	fun serializeModel(os: OutputStream, model: Model) {
		serialize(os, model)
	}

	/**
	 * Serialize core model to file
	 *
	 * @param model core model
	 * @param file  file
	 * @throws IOException io exception
	 */
	@JvmStatic
	@Throws(IOException::class)
	fun serializeCoreModel(model: CoreModel, file: File) {
		FileOutputStream(file).use { os ->
			serializeCoreModel(os, model)
		}
	}

	/**
	 * Serialize core model to output stream
	 *
	 * @param model core model
	 * @param os    output stream
	 * @throws IOException io exception
	 */
	@Throws(IOException::class)
	fun serializeCoreModel(os: OutputStream, model: CoreModel?) {
		serialize(os, model)
	}

	/**
	 * Serialize object to output stream
	 *
	 * @param os     output stream
	 * @param object object
	 * @throws IOException io exception
	 */
	@Throws(IOException::class)
	private fun serialize(os: OutputStream, `object`: Any?) {
		ObjectOutputStream(os).use { oos ->
			oos.writeObject(`object`)
		}
	}
}
